pkg/asset/manifests/dns.go

```diff
@@ -244,31 +256,44 @@ func GetProjectForDNSZones(installConfig *installconfig.InstallConfig) string {
 // GetGCPPrivateZoneName attempts to find the name of the private zone for GCP installs. When a shared vpc install
 // occurs, a precreated zone may be used. If a zone is found (in this instance), then the zone should be paired with
 // the network that is supplied through the install config (when applicable).
-func GetGCPPrivateZoneName(ctx context.Context, client *icgcp.Client, installConfig *installconfig.InstallConfig, clusterID string) (string, error) {
-	privateZoneID := fmt.Sprintf("%s-private-zone", clusterID)
+func GetGCPPrivateZoneName(ctx context.Context, client *icgcp.Client, installConfig *installconfig.InstallConfig, clusterID string) (string, bool, error) {
+	defaultPrivateZoneID := fmt.Sprintf("%s-private-zone", clusterID)
+	privateZoneID := defaultPrivateZoneID
+	shouldCreateZone := true
+
 	if installConfig.Config.GCP.NetworkProjectID != "" {
 		project := GetProjectForDNSZones(installConfig)
```

Comment:
Another instance where a common method that returns both the project and zone would be handy.